pkg/server/test/write_authzmodel.go

```diff
@@ -455,13 +472,7 @@ type other
 				),
 				TypeDefinitions: parser.MustTransformDSLToProto(`model
   schema 1.1
-type user
-
-type other
-  relations
-	define x: [user] but not y
-	define y: [user] but not z
-	define z: [user] or x`).TypeDefinitions,
```

Comment:
this test `validate_model_size` can use a much simpler model and it will still fail.

Reply:
Tread lightly in change existing tests if possible, because a lot of these were added in response to community (or customer) related issues. I'd lean on the side of adding instead of repurposing.

Reply:
> I will re-verify that I didn't miss anything while marking them as "TODO remove".

Did this. Barring some incorrect comments, all the tests were copied over correctly. 

At the end of the day, we only need one test for `NewAndValidate` that ensures that it is calling `HasNoEntrypoints`. And we only need one test in `WriteAuthorizationCommand` that ensures that it is calling `NewAndValidate`.

Reply:
So until we have that we should make sure we're testing at the higher level, because `WriteAuthorizationCommand` isn't guaranteed to call `NewAndValidate`, and assuming that it does is where the bug can creep in. What matters is the correctness of `WriteAuthorizationCommand`, however it achieves it. `NewAndValidate` is just the proxy.

Reply:
Would it be OK if in the same PR where i remove the TODOs i introduce interfaces for those? It is wrong to have a test for `NewAndValidate` inside a test for `WriteAuthorizationCommand` because then, other places that rely on `NewAndValidate` working correctly will be unprotected

Reply:
Let's table this discussion for now 😆